### ISSUE TITLE
How to choose the loss function for Binary Autoencoders

### ISSUE THREAD

**Issue by kingfengji** [fchollet/keras#2143]
Hi all,

I wish to train a stacked auto-encoder, but the input vectors*\* **are binary, ie [0,0,1,1,0,1..1], and they are quite sparse......****

so picking the loss function as mse doesnt work so well, I wish to know 
1) will it makes sense  changing the last layer of decoder into a **softmax activation **
2) which **loss function** should I use.

Thanks

**Comment by lemuriandezapada** [comment]
binary_crossentropy

**Comment by kingfengji** [comment]
@lemuriandezapada 
no, it is not a one-hot vector, it has multiple ones in it.

**Comment by pengpaiSH** [comment]
@kingfengji How did you finally fix that problem?
